tools/cmake/toolchains/cbmc.cmake

```diff
@@ -0,0 +1,109 @@
+# ``````````````````````````````````````````````````````````````````````````````
+#               Toolchain for building and running CBMC proofs
+#               ==============================================
+#
+# CBMC proofs require include paths, compile definitions, and other variables
+# that are common to all proofs. At the same time, individual proofs may wish to
+# extend these options---for example, including headers only needed by that
+# proof, or proving more than just memory safety.
+#
+# This file sets up variables that can be appended to in the CMakeLists.txt of
+# each proof. Most of the recipes for building and running proofs are written in
+# the cbmc_recipes.cmake file. cbmc_recipes.cmake gets included at the bottom of
+# each proof, and uses the variables defined in this toolchains file. This file
+# is only processed once when using the CBMC toolchain, while cbmc_recipes.cmake
+# gets included separately by each proof.
+# ______________________________________________________________________________
+
+set(CBMC 1)
+
+#  0 no output
+#  1 errors
+#  2 warnings
+#  4 results
+#  6 status information
+#  8 statistical information
+#  9 progress information
+# 10 debug info
+set(CBMC_VERBOSITY 4 CACHE STRING "Verbosity of CBMC and related tools")
+
+# ______________________________________________________________________________
+#     Variables that are not intended to be directly changed by proof-writers
+# ``````````````````````````````````````````````````````````````````````````````
+
+# The list of all proof names (used to check that proof names are unique)
+set(cbmc_proof_names "" CACHE INTERNAL "")
+
+# The cbmc root
+set(cbmc_dir ${CMAKE_SOURCE_DIR}/tools/cbmc)
```

Comment:
These dir names are FreeRTOS and MQTT specific, crypto uses different stuff.

Reply:
As agreed, let's merge this and refactor it later once it's clear what other projects will need